python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -183,14 +185,44 @@ def execute_action(self, data):
             logger.error("Command failed: %s", args)
             logger.error("Failed command stdout: %s", stdout)
             logger.error("Failed command stderr: %s", stderr)
-            self.host.report_error("%s\n\n%s" % (stdout, stderr))
+            self.host.reply(
+                dict(
+                    retcode=constants.COMMAND_FAILED,
+                    out=stdout,
+                    err=stderr,
+                )
+            )
             return
 
+        filtered_output = []
+
+        # We need to filter stdout before we send it to the client.
+        # We look for lines that we know came from the custom log
+        # handler that the execute_command script builds, and we
+        # remove that header from those lines and keep them so that
+        # they're passed up to the client.
+        tag = "SGTK:"
+        tag_length = len(tag)
+
+        # We check both stdout and stderr. We identify lines that start with
+        # our tag, and if we find one, we remove the tag, and then base64
+        # decode the rest of the message. The message is encoded this way
+        # because it collapses multi-line log messages into a single line
+        # of text. This is important, because there's only one tag per log
+        # message, and we would otherwise filter out everything in the log
+        # message that wasn't on the first line of text before any newlines.
+        for line in stdout.split("\n") + stderr.split("\n"):
+            if line.startswith(tag):
+                filtered_output.append(base64.b64decode(line[tag_length:]))
+
+        filtered_output_string = "\n".join(filtered_output)
+        logger.debug("Filtered and decoded log output: %s", filtered_output_string)
```

Comment:
Maybe a bit much since it'll all be in tk-shotgun.log. I had put that in there while debugging just so I had a single log file to tail, but I think it's worth removing this for release.